examples/free-identifier-case.kl

```diff
@@ -2,48 +2,31 @@
 
 (import "let.kl")
 (import (shift "prelude.kl" 1))
-(import (shift "quasiquote.kl" 1))
-
-(meta
-  (define identifier?
-    (lambda (x)
-      (syntax-case x
-        [(ident x) (true)]
-        [_ (false)]))))
+(import (shift "identifier.kl" 1))
+(import (shift "dot-dot-dot.kl" 1))
 
 (define-macros
   ([free-identifier-case
     (lambda (stx)
-      (syntax-case stx
-        [(cons _ body)
-         (syntax-case body
-           [(cons scrut cases)
-            (pure
-             (quasiquote/loc stx
-               (let (x ,scrut) (free-identifier-case-aux x ,cases))))])]))]
+      (syntax-case stx ()
+        [(_ ,scrut ,cases ...)
+         (pure (replace-loc stx
+                 `(let (x ,scrut) (free-identifier-case-aux x ,cases ...))))]))]
```

Comment:
Here, I slightly tweaked the syntax of `free-identifier-case-aux`: a call now looks like

    (free-identifier-case-aux (just 3)
      [(just x) x]
      [(nothing) 0])

instead of

    (free-identifier-case-aux (just 3)
      ([(just x) x]
       [(nothing) 0]))

I suspect the reason for these seemingly-spurious extra parentheses is simply that it was not easy to construct the recursive call using `quasiquote/loc`. Since `fancy-quasiquote` now makes it easy, I removed the spurious parentheses.

Reply:
Speaking of `quasiquote/loc`: I considered writing a `fancy-quasiquote/loc` around `fancy-quasiquote`, but since `fancy-quasiquote` can be invoke very easily using a single backtick character, there didn't seem to be much value over simply calling `replace-loc` around the result of a `fancy-quasiquote` call.